Add a section on auto_download to the readme

psykzz opened the pull request:
### Description of the Change
Adds context on `--auto_download` to the readme
Additionally, on non-windows systems, this will try to use `7z` from PATH if available.

### Benefits
Just explains usage.
Extends non-windows usage

### Applicable Issues
N/A

psykzz committed:
Add a section on auto_download to the readme

lutzmor commented on README.md:
```diff
@@ -28,9 +28,10 @@ All scripts are run built with python3, but should be able to be run with python
+1. To run with a local simc you have three options:
 - use the simc you have on your Path ( nothing to setup here )
 - use a simc located in a separated folder, create a `local_secrets.py` inside the root directory and set `simc_path = '{"nightly": "path/to/executable", "rework": "another/path"}'`
+- add the `--auto_download` parameter to download the latest simc nightly build into the `auto_download/` folder. This will override any choice in `config.yml` with the path to the newly downloaded version.
```
Maybe its worth mentioning here that this --auto_download option is only possible within windows based systems ;). Because i don't think "C:\.....7zip.exe" is a good way to try unpack stuff under linux ;).

psykzz replied:
Good point for sure, if needed i could add support for linux/osx. I just wasn't sure of the usage.

lutzmor replied:
I will take a look at this :) i only use linux so maybe i can help here :)

lutzmor replied:
Hmm i think for the moment we can't support linux here only osx would be possible because simc is not build for linux seperate :). And osx i currently don't have a installation available :).

psykzz replied:
Alright, will revert the small change and add the note win64 only is supported.

lutzmor replied:
Yeah that would be perfect i think thats the best for the moment :), if i find a mac somewhere around me i will try to get it working there as well. Thanks!

scamille replied:
Supporting different platforms not only requires adjusting 7z, but also the architecture name of the simc nightly file.

Might be nice to support macos, but you could also just hard-check for os windows in python, and throw a error if someone uses it on a unsupported platform.

lutzmor replied:
@scamille like i mentioned at the other place its 7z is not the way to go for mac and linux. I can't test all this because i only have linux at home ;). Maybe create a issue for supporting macos, so we don't make the conversation here any longer :). For the moment i think the notice is okay, and the support for macos can be implemented in a seperate pull request.

psykzz committed:
Add support for linux/osx in auto_download

This only checks for a default install of 7z in path, but should cover typical cases

psykzz committed:
Revert osx, and add note

lutzmor commented on internal/auto_download.py:
```diff
@@ -11,7 +11,7 @@
 def download_latest():
     """main download function"""
-    seven_zip_paths = ["7z.exe", "C:/Program Files/7-Zip/7z.exe"]
+    seven_zip_paths = ["7z.exe", "C:/Program Files/7-Zip/7z.exe"] if os.name == 'nt' else ['7z']
```
Basicly 7z is not neccessary on anything beside windows, on OSX its a dmg file which needs to be mounted and then i think something needs to get copied around and unter linux its unzip or something like tar if simc would be packed as a *.tar.gz. Like mentioned at the other place linux isn't available and osx i'm unsure how to try it :)

psykzz replied:
Yeah additionally the download only looks at the latest win64 file too, i was being dumb thinking this was all that was involved.
